cogs/handover.py

```diff
@@ -0,0 +1,180 @@
+                ":warning:",
+            )
+            return
+
+        for category in main_guild.categories:
+            if "committee" in category.name.lower() and "archive" not in category.name.lower():
+                await initial_response.edit(
+                    content=f":hourglass: Updating channels in category: {category.name} "
+                    ":hourglass:",
+                )
+                for channel in category.channels:
+                    logger.debug("Resetting channel permissions for channel: %s", channel)
+                    await channel.set_permissions(committee_elect_role, overwrite=None)
+
+        await initial_response.edit(
+            content=":hourglass: Giving committee members access to #handover and "
+            "removing committee role... :hourglass:",
+        )
+
+        for member in committee_role.members:
+            logger.debug("Giving user: %s, access to #handover", member)
+            await handover_channel.set_permissions(
+                member,
+                read_messages=True,
+                send_messages=True,
+            )
+
+            logger.debug("Removing committee role from user: %s", member)
+            await member.remove_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(
+            content=":hourglass: Giving committee-elect committee role and "
+            "removing committee-elect... :hourglass:",
+        )
+
+        for member in committee_elect_role.members:
+            logger.debug("Giving user: %s, the committee role.", member)
+            await member.add_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+            logger.debug("Removing committee-elect role from user: %s", member)
+            await member.remove_roles(
+                committee_elect_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(content=":white_check_mark: Handover procedure complete!")
+
+
+class ResetRolesCommandCog(TeXBotBaseCog):
+    """Cog class that defines the reset_roles command."""
+
+    @discord.slash_command( # type: ignore[no-untyped-call, misc]
+        name="reset_roles",
+        description="Resets member and year roles",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def reset_roles(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "reset_roles" command.
+
+        The "reset_roles" command removes the memeber and respective year roles from
+        anyone that has them and subsequently resets the GroupMadeMember database Model.
+        """
```

Comment:
Include note comment on why accessors are at the top of the function (to aid new developers in the future):

```python
# NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
```